app/furniture/marketplace/delivery_areas/index.html.erb

```diff
@@ -5,8 +5,9 @@
     <div class="mt-3 grid grid-cols-1 gap-5 sm:gap-6 sm:grid-cols-2 lg:grid-cols-4">
     <% delivery_areas.each do |area| %>
       <%= render CardComponent.new(classes: "py-2") do %>
-        <%= area.label %>
-        <%= number_to_currency(area.price) %>
+        <%= link_to marketplace.location(:edit, child: :delivery_area) do %>
+          <%= area.label %>: <%= number_to_currency(area.price) %>
+        <%- end %>
```

Comment:
Route issues on page render.
<img width="1680" alt="Screenshot 2023-04-02 at 9 56 09 PM" src="https://user-images.githubusercontent.com/16140873/229415643-86ce05e9-cfa4-47c1-8f0a-bc3d605e6914.png">


Yet the route totally exists!

```
   room_marketplace_delivery_areas GET    /rooms/:room_id/marketplaces/:marketplace_id/delivery_areas(.:format)                                         marketplace/delivery_areas#index
          edit_room_marketplace_delivery_area GET    /rooms/:room_id/marketplaces/:marketplace_id/delivery_areas/:id/edit(.:format)                                marketplace/delivery_areas#edit
               room_marketplace_delivery_area GET    /rooms/:room_id/marketplaces/:marketplace_id/delivery_areas/:id(.:format)                                     marketplace/delivery_areas#show
```

Reply:
I think you're looking for `area.location(:edit)`, since you have access to the "leaf" of the object graph (a delivery area), and you're looking for the `edit` action for that leaf.